src/Generation/Generator/Renderer/Public/ParameterToNativeExpression/Converter/String.cs

```diff
@@ -21,12 +21,22 @@ public void Initialize(ParameterToNativeData parameter, IEnumerable<ParameterToN
         if (parameter.Parameter.Direction == GirModel.Direction.InOut)
             throw new NotImplementedException($"{parameter.Parameter.AnyType}: String type with direction=inout not yet supported");
 
-        var prefix = parameter.Parameter.Direction == GirModel.Direction.Out
-            ? "out "
-            : string.Empty;
+        // TODO - support output strings
+        if (parameter.Parameter.Direction == GirModel.Direction.Out)
+            throw new NotImplementedException($"{parameter.Parameter.AnyType}: String type with direction=out not yet supported");
 
         var parameterName = Parameter.GetName(parameter.Parameter);
         parameter.SetSignatureName(parameterName);
-        parameter.SetCallName(prefix + parameterName);
+
+        string nativeVariableName = Parameter.GetConvertedName(parameter.Parameter);
+        var stringType = (GirModel.String) parameter.Parameter.AnyType.AsT0;
+
+        string createExpression = $"{Model.String.GetInternalOwnedHandleName(stringType)}.Create({parameterName})";
+        if (parameter.Parameter.Nullable)
+            createExpression = $"{parameterName} is null ? {Model.String.GetInternalNullHandleName(stringType)}.Instance : {createExpression}";
+
+        parameter.SetExpression($"{Model.String.GetInternalHandleName(stringType)} {nativeVariableName} = {createExpression};");
```

Comment:
Split file into Platform / Utf8 version and use the corresponding utf8 / platform model to set the expression.